scripts/src/java/org/oppia/android/scripts/testfile/TestFileCheck.kt

```diff
@@ -23,44 +23,79 @@ fun main(vararg args: String) {
 
   val testFileExemptiontextProto = "scripts/assets/test_file_exemptions"
 
-  // A list of all the files to be exempted for this check.
-  // TODO(#3436): Develop a mechanism for permanently exempting files which do not ever need tests.
-  val testFileExemptionList = loadTestFileExemptionsProto(testFileExemptiontextProto)
-    .getExemptedFilePathList()
-
-  // A list of all kotlin files in the repo to be analyzed.
-  val searchFiles = RepositoryFile.collectSearchFiles(
-    repoPath = repoPath,
-    expectedExtension = ".kt",
-    exemptionsList = testFileExemptionList
-  )
-
-  // A list of all the prod files present in the repo.
-  val prodFilesList = searchFiles.filter { file -> !file.name.endsWith("Test.kt") }
-
-  // A list of all the test files present in the repo.
-  val testFilesList = searchFiles.filter { file -> file.name.endsWith("Test.kt") }
-
-  // A list of all the prod files that do not have a corresponding test file.
-  val matchedFiles = prodFilesList.filter { prodFile ->
-    !testFilesList.any { testFile ->
-      testFile.name == computeExpectedTestFileName(prodFile)
-    }
+  val testFileCheck = TestFileCheck(repoPath, testFileExemptiontextProto)
+  testFileCheck.execute()
+}
+
+/**
+ * Class for checking the presence of test files in a repository.
+ *
+ * @param repoPath The path of the repo
+ * @param testFileExemptiontextProto the location of the test file exemption textproto file.
```

Comment:
```suggestion
 * @param testFileExemptiontextProto the location of the test file exemption textproto file
```

Reply:
Recent changes removed the function itself, so these are no longer available.